nifi-nar-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/AMQPUtils.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.amqp.processors;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.ProcessSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rabbitmq.client.AMQP.BasicProperties;
+
+/**
+ * Utility helper class simplify interructions with target AMQP API and NIFI
+ * API.
+ *
+ */
+abstract class AMQPUtils {
+
+    public final static String AMQP_PROP_DELIMITER = "$";
+
+    public final static String AMQP_PROP_PREFIX = "amqp" + AMQP_PROP_DELIMITER;
+
+    private final static Logger logger = LoggerFactory.getLogger(AMQPUtils.class);
+
+    private final static List<String> propertyNames = Arrays.asList("amqp$contentType", "amqp$contentEncoding",
```

Comment:
Consider yes, but let's not do it for this release. The user can still have UpdateAttributes upstream and map mime.type to contentType. Just want to make sure we avoid potential for collisions until we start getting some usage out of it